### DISCUSSION TITLE
what availability of erasure coding？

### DISCUSSION THREAD

**Discussion by lddlww** [cubefs/cubefs#3485] [Q&A] [answered]
my blobnode is 4 ， per blobnode have 10 disks

how can i compute availability of my erasure coding structure？for EC12P4，EC3P3，EC6P6

in other ways， what max number of blobnodes can down， if  my cubefs can put and get？

**Comment by FourSpaces** [comment, marked as answer]
Your number of Blobnode nodes is too small, a reasonable number of nodes should be a multiple of N+M+l.
At present, it seems that with good luck, you may lose at most one of the configurations above, and you can continue to write and read data. In a 4-node Blobnode, losing one machine may result in the node being non writable but readable.
 
If you are limited by your current hardware resources, you can use EC3P3 and modify the configuration of access. conf by adding the configuration under the stream configuration item, which can ensure that losing one node can continue reading and writing, and losing two nodes can only read.
```
"stream": {
....
"code_mode_put_quorums": {"11": 4}
}
```

**Reply by lddlww** [reply]
that's explained very clearly，thanks
